src/org/myworldgis/netlogo/LoadDataset.java

```diff
@@ -157,25 +217,56 @@ private static VectorDataset loadGeoJson (String geojsonFilePath,
             } catch (org.json.simple.parser.ParseException e){
                 throw new ExtensionException("Error parsing " + geojsonFilePath);
             }
+
+            String[] propertyNames;
+            VectorDataset.PropertyType[] propertyTypes;
+
+            if (reader.getShouldAddZField()) {
+                propertyNames = new String[reader.getPropertyNames().length + (reader.getShouldAddZField() ? 1 : 0)];
```

Comment:
Is this inner `reader.getShouldAddZField()` necessary?  Looks like it'll always be true from the if condition above.